app/models/host/managed.rb

```diff
@@ -335,7 +335,8 @@ def handle_realm
   def diskLayout
     @host = self
     template = disk.blank? ? ptable.layout : disk
-    pxe_render(template.tr("\r", ''))
+    template_name = disk.blank? ? ptable.name : 'Custom disk layout'
```

Comment:
Thanks, that's one half, if you could also add a test for when a ptable is assigned that'll cover the other code path.